src/app/shared/services/search.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+import { Movies } from '@shared/interfaces/movies';
+import { APIURL } from '@shared/apiUrl';
+import { Observable, of } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { HttpClient } from '@angular/common/http';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class SearchService {
+
+  // constructor(private http: HttpClient) { }
+  // private apiUrl: Object = APIURL;
+  // movies$: Observable<object>;
+
+
+  // searchMovies(term: string): Observable<Movies<Object>> {
+  //   if (!term.trim()) {
+  //     // if not search term, return empty hero array.
+  //     return of(null);
+  //   }
+  //   let formUrl = `${this.apiUrl['root'] + this.apiUrl['URLS']['search-movie'] + this.apiUrl['api_key'] + '&query=' + term.toLowerCase()}`;
+
+  //   return this.http.get<Movies<Object>>(formUrl)
+  //     .pipe(
+  //       catchError(this.handleError<Movies<Object>>('searchHeroes', []))
+  //     );
+
+  // }
+
+  // passSearchData(movies: Observable<object>): void {
+  //   this.movies$ = movies;
+  //   this.movies$.subscribe(s => console.log(s))
```

Comment:
you shouldn`t subscribe in the service. Subscriptions should be in the component.

Reply:
I'm not using the search service at all. I'll delete it now in order to avoid further confusion.